packages/adblocker/src/resources.ts

```diff
@@ -63,56 +96,47 @@ export default class Resources {
   }
 
   public static parse(data: string, { checksum }: { checksum: string }): Resources {
-    const typeToResource: Map<string, Map<string, string>> = new Map();
-    const trimComments = (str: string) => str.replace(/^\s*#.*$/gm, '');
-    const chunks = data.split('\n\n');
-
-    for (const chunk of chunks) {
-      const resource = trimComments(chunk).trim();
-      if (resource.length !== 0) {
-        const firstNewLine = resource.indexOf('\n');
-        const split = resource.slice(0, firstNewLine).split(/\s+/);
-        const name = split[0];
-        const type = split[1];
-        const body = resource.slice(firstNewLine + 1);
-
-        if (name === undefined || type === undefined || body === undefined) {
-          continue;
-        }
+    const distribution: ResourcesDistribution = JSON.parse(data);
 
-        let resources = typeToResource.get(type);
-        if (resources === undefined) {
-          resources = new Map();
-          typeToResource.set(type, resources);
-        }
-        resources.set(name, body);
+    const js: Map<string, string> = new Map();
+    const resources: Map<string, Resource> = new Map();
+
+    for (const redirect of distribution.redirects) {
+      const body =
+        redirect.encoding === 'base64' ? atobPolyfill(redirect.content) : redirect.content;
+
+      for (const name of redirect.names) {
+        resources.set(name, {
+          contentType: redirect.contentType,
+          body,
+        });
       }
     }
 
-    // The resource containing javascirpts to be injected
-    const js: Map<string, string> = typeToResource.get('application/javascript') || new Map();
-    for (const [key, value] of js.entries()) {
-      if (key.endsWith('.js')) {
-        js.set(key.slice(0, -3), value);
+    for (const scriptlet of distribution.scriptlets) {
+      // Compile content with dependencies
+      let content = '';
+
+      for (const dependencyName of scriptlet.dependencies) {
+        const dependency = distribution.scriptlets.find((scriptlet) =>
+          scriptlet.names.includes(dependencyName),
+        );
+        if (dependency !== undefined) {
+          content += dependency.content + '\n\n';
+        }
```

Comment:
was hoping we can resolve the dependencies on runtime now that we have them at hand - but ok, in this version we can have it at parse time to avoid additional complexity. 

btw, you have to check dependencies recursively and add the global variable to shared state

Reply:
The very first version of dependency resolving completed in [`5e84b73` (#4270)](https://github.com/ghostery/adblocker/pull/4270/commits/5e84b733157fb2c579e164dc5c924452cef2303a). We can improve this time to time.